src/App.tsx

```diff
@@ -8,7 +7,58 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [select, setSelect] = useState('all');
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [selectTodo, setSelectTodo] = useState<Todo>();
+
+  useEffect(() => {
+    getTodos().then(result => setAllTodos(result));
+
+    setVisibleTodos(allTodos.filter(todo => {
+      switch (select) {
+        case ('all'):
+          return todo.title.toUpperCase().includes(query.toUpperCase().trim());
+
+        case ('active'):
+          return todo.title.toUpperCase().includes(query.toUpperCase().trim())
+            && !todo.completed;
+
+        case ('completed'):
+          return todo.title.toUpperCase().includes(query.toUpperCase().trim())
+            && todo.completed;
+
+        default:
+          return false;
+      }
+    }));
+  }, [query, select, allTodos]);
+
+  const onQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const onSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelect(event.target.value);
+  };
+
+  const onButtonClick = () => {
+    setQuery('');
+  };
+
+  const gettingTodo = (todo: Todo) => {
+    setSelectTodo(todo);
+  };
+
+  const onModalClose = () => {
+    setSelectTodo(undefined);
```

Comment:
It's better to use null instead of undefined
```suggestion
    setSelectTodo(null);
```